cpp/src/join/mixed_join_semi.cu

```diff
@@ -151,15 +155,20 @@ std::unique_ptr<rmm::device_uvector<size_type>> mixed_join_semi(
   // TODO: The non-conditional join impls start with a dictionary matching,
   // figure out what that is and what it's needed for (and if conditional joins
   // need to do the same).
-  auto& probe                  = swap_tables ? right_equality : left_equality;
-  auto& build                  = swap_tables ? left_equality : right_equality;
-  auto probe_view              = table_device_view::create(probe, stream);
-  auto build_view              = table_device_view::create(build, stream);
-  auto left_conditional_view   = table_device_view::create(left_conditional, stream);
-  auto right_conditional_view  = table_device_view::create(right_conditional, stream);
-  auto& build_conditional_view = swap_tables ? left_conditional_view : right_conditional_view;
-  row_equality equality_probe{
-    cudf::nullate::DYNAMIC{has_nulls}, *probe_view, *build_view, compare_nulls};
+  auto& probe                 = swap_tables ? right_equality : left_equality;
```

Comment:
Why do we have `swap_tables`?

Reply:
The smaller table is used as the `build` table. Since `mixed_join` doesn't have a stateful API like `hash_join`, you see the swapping happening at every step whereas `hash_join` does the swapping in its constructor/outer API